### ISSUE TITLE
Prevent double click likes 

### ISSUE THREAD

**Issue by micke-a** [turboDi/grails-likeable#2]
Hi

I recently needed to implement like functionality and installed this plug-in, did the trick. Thank you!

Recently discovered that users have double clicked my ajaxy widget which performs a http request to the likeable/like endpoint provided by the plug-in. 
This results in NonUniqueResultException: query did not return a unique result: 2 errors.

First off a question is there an easy way to prevent this without plug-in changes? I could implement my own endpoint with protection but prefer to keep using the plug-in provided functionality.

Second it would be brilliant if the plug-in could be updated to be more forgiving so that it doesn't throw an exception when this happens and/or prevent records with the same liker and liked ids from being persisted.

Thanks again for the plug-in!

- Micke

**Comment by turboDi** [comment]
Hi, Micke

Glad to hear that the plugin helped someone but me!
Could you point the plugin version you are using? I've added a constraint to prevent multiple likes from the same person since 0.3.0. 
Also in this version I've marked Likeable controller as deprecated in favor of new LikeController which follows RESTful principles (because I believe that RESTful approach is more suitable for like functionality). You can look into it or you can stay with the old one.

Feel free to ask if any questions

**Comment by micke-a** [comment]
Hi,
I'm using compile 0.2.0 .

I'll checkout version 0.3.0 , I'll mark this issue as closed!

Many thanks,
Micke

**Comment by micke-a** [comment]
Fyi, just tried and get this
```
| Error Compilation error: startup failed:
C:\Users\User\.grails\2.2.4\projects\gapet\plugins\likeable-0.3.0\grails-app\controllers\ru\jconsulting\likeable\LikeController.groovy: 3: unable to resolve class grails.rest.RestfulController
 @ line 3, column 1.
   import grails.rest.RestfulController
   ^
C:\Users\User\.grails\2.2.4\projects\gapet\plugins\likeable-0.3.0\grails-app\controllers\ru\jconsulting\likeable\LikeController.groovy: 10: unable to resolve class grails.rest.RestfulController 
 @ line 10, column 1.
   class LikeController extends RestfulController<Like> {
```

Looks like my 2.2.4 grails version is to old for version 0.3.0

**Comment by turboDi** [comment]
Hm, It seems I forgot to increment grails version requirements in plugin def file...
However, I'm going to fix 0.2.0 soon. Or you can, pull requests are welcome)
For now you can prevent double post in your widget (for example disable button until server responds)
